### PR TITLE
:sparkles: add search bar

### PR DESCRIPTION
**Pull request by issa-diallo**
- An enhancement is planned #185
    for filter by patient in list prescription
    
- Fetch patient and switch to parent component and
    patient is switched to props for each prescription
    
-Added search bar for patients and prescriptions

### CODE REVIEW

File: src/pages/PatientsPage.tsx [hunk 1]
```diff
@@ -5,11 +5,32 @@ import usePatients from '../hook/patient.hook'
...
 
+  // Search Bar
+  const [filteredPatients, setFilteredPatients] = useState<Patient[]>(patients)
+
+  const filterByPatients = (patient: Patient, searchValue: string): boolean => {
+    const fullName = `${patient.lastname} ${patient.firstname}`.toLowerCase()
+    return (
+      fullName.includes(searchValue.toLowerCase()) ||
+      patient.lastname.toLowerCase().includes(searchValue.toLowerCase()) ||
+      patient.firstname.toLowerCase().includes(searchValue.toLowerCase()) ||
```
Comment (AndreMiras) [inline src/pages/PatientsPage.tsx:24]: Why do you need to search in lastname and firstname if you're already searching in fullname?

File: src/pages/PatientsPage.tsx [hunk 2]
```diff
@@ -5,11 +5,32 @@ import usePatients from '../hook/patient.hook'
...
 
 const PatientsPage = (): JSX.Element => {
   const { patients, reloadPatients } = usePatients()
   const { t } = useTranslationHook()
 
+  // Search Bar
+  const [filteredPatients, setFilteredPatients] = useState<Patient[]>(patients)
+
+  const filterByPatients = (patient: Patient, searchValue: string): boolean => {
+    const fullName = `${patient.lastname} ${patient.firstname}`.toLowerCase()
```
Comment (AndreMiras) [inline src/pages/PatientsPage.tsx:20]: DRY `searchValue.toLowerCase()`

File: src/components/Prescriptions.tsx [hunk 3]
```diff
@@ -18,6 +21,56 @@ const Prescriptions: FunctionComponent<PrescriptionsProps> = ({
...
+
+        patientResults.forEach(({ id, patient }) => {
+          patientsData[id] = patient
+        })
+
+        setPatients(patientsData)
+      } catch (error) {
+        console.error(error)
+      }
+    }
```
Comment (AndreMiras) [inline src/components/Prescriptions.tsx:55]: So I imagine this is in preparation for "prescription full text search"

**Review by AndreMiras — COMMENTED** [review]
Left a few comments

File: src/pages/PatientsPage.tsx (hunk 1 above)
Comment (issa-diallo) [reply src/pages/PatientsPage.tsx:24]: depend on you feel you're starting the search

File: src/components/Prescriptions.tsx (hunk 3 above)
Comment (issa-diallo) [reply src/components/Prescriptions.tsx:55]: exactly

File: src/pages/PatientsPage.tsx (hunk 1 above)
Comment (issa-diallo) [reply src/pages/PatientsPage.tsx:24]: yes I see

File: src/pages/PatientsPage.tsx (hunk 2 above)
Comment (issa-diallo) [reply src/pages/PatientsPage.tsx:20]: I don't see DRY

**Review by AndreMiras — APPROVED** [review]

**Pull request merged**